ssh/server/server.go

```diff
@@ -37,9 +40,65 @@ func NewServer(opts *Options, tunnel *httptunnel.Tunnel) *Server {
 	}
 
 	server.sshd = &gliderssh.Server{ // nolint: exhaustruct
-		Addr:             ":2222",
-		PasswordHandler:  auth.PasswordHandler(tunnel),
-		PublicKeyHandler: auth.PublicKeyHandler(tunnel),
+		Addr: ":2222",
+		ConnCallback: func(ctx gliderssh.Context, conn net.Conn) net.Conn {
+			ctx.SetValue("conn", conn)
+
+			return conn
+		},
+		BannerHandler: func(ctx gliderssh.Context) string {
```

Comment:
You could optimize the repetitive `log.Fields` calls by creating the log object once and reusing it throughout the function.

Approach A:

```go
logFields := log.Fields{
    "uid":   ctx.SessionID(),
    "sshid": ctx.User(),
}

log.WithFields(logFields).Error("some error")
```

Approach B:

 ```
 ```go
logEntry := log.WithFields(log.Fields{
    "uid":   ctx.SessionID(),
    "sshid": ctx.User(),
})

logEntry.Error("some error")
```

Reply:
I prefer `B` :-D

Reply:
Done

Reply:
> I prefer `B` :-D

Me too.

One advantage of Approach B is that it allows for the addition of more fields if necessary. By maintaining a base log entry for the entire function, and if needed, each entry can add more fields.